keyboards/takashicompany/spreadwriter/spreadwriter.c

```diff
@@ -0,0 +1,28 @@
+// Copyright 2022 takashicompany (@takashicompany)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "spreadwriter.h"
+
+#ifdef ENCODER_ENABLE
+bool encoder_update_kb(uint8_t index, bool clockwise) {
+    if (!encoder_update_user(index, clockwise)) return false;
+
+    keypos_t key;
+    if (index == 0) {
+        if (clockwise) {
+            key.row = 4;
+            key.col = 0;
+        } else {
+            key.row = 0;
+            key.col = 6;
+        }
+        uint8_t layer = layer_switch_get_layer(key);
+        uint16_t keycode = keymap_key_to_keycode(layer, key);
+
+        tap_code16(keycode);
+        return false;
+    }
+
+    return true;
+}
+#endif
```

Comment:
Two issues here: 
1. tap_code16 only deals with basic and modded keycodes.  It WILL not handle quantum keycodes or macros.  
2. Develop (which lands at the end of month) includes an "encoder map" feature, which allows for a keymap like array that fully supports every keycode.  https://qmk.github.io/qmk_docs_devel/#/feature_encoders?id=encoder-map

If you still wish to implement this, then something like this would work better:
https://github.com/drashna/qmk_firmware/blob/78fbc54d83d57b830a51bb52cb86daa394d46baf/keyboards/jones/v1/keymaps/via/keymap.c#L168-L207

However, this should be one exclusively for the via keymap.

Reply:
Thanks. 
Done : acaf6ca